File: src/main/java/ru/spbau/mit/ir/crawler/Website.java
package ru.spbau.mit.ir.crawler;

import com.panforge.robotstxt.RobotsTxt;

import java.io.InputStream;
import java.net.URL;

public class Website {
    private String rootPage;

    public Website(String rootPage) {
        this.rootPage = rootPage;
    }

    public String nextUrl() {
        return rootPage;
    }

    public boolean permitsCrawl(String userAgent, String url) {
        try {
            URL link = new URL(url);
            String host = link.getHost();
            String protocol = link.getProtocol();
            String path = protocol + "://" + host + "/robots.txt";
            try (InputStream robotsTxtStream = new URL(path).openStream()) {
                RobotsTxt robotsTxt = RobotsTxt.read(robotsTxtStream);
                boolean hasAccess = robotsTxt.query(userAgent, url);
                return hasAccess;
            }
        } catch (Exception e) {
            e.printStackTrace();
            return false;
        }
    }
}

File: src/main/kotlin/ru/spbau/mit/ir/crawler/Website.kt
package ru.spbau.mit.ir.crawler

import com.panforge.robotstxt.RobotsTxt
import java.io.IOException

import java.io.InputStream
import java.net.URL

class Website(private val rootPage: String) {

    fun nextUrl() = rootPage

    fun permitsCrawl(userAgent: String, link: URL): Boolean {
        val host = link.host
        val protocol = link.protocol
        val path = "$protocol://$host/robots.txt"

        try {
            URL(path).openStream().use { robotsTxtStream ->
                val robotsTxt = RobotsTxt.read(robotsTxtStream)
                return robotsTxt.query(userAgent, link.toExternalForm())
            }
        } catch (e: IOException) {
            e.printStackTrace() // todo: logging?
            return false
        }

    }
}
